core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java

```diff
@@ -512,7 +543,51 @@ void createMeasurementEvent(
                             throw CairoException.instance(0).put("queue buffer overflow");
                         }
                         bufPos += colNameLen;
+
+                        final CharSequence columnName;
+                        if (protoParser.hasNonAsciiChars()) {
+                            final StringSink tempSink = localDetails.tempSink;
+                            tempSink.clear();
+                            //we need to convert UTF8 encoding because column comparison is case insensitive
+                            if (!Chars.utf8Decode(entity.getName().getLo(), entity.getName().getHi(), tempSink)) {
+                                throw CairoException.instance(0).put("invalid UTF8 in value for ").put(entity.getName());
+                            }
+
+                            columnName = tempSink.toString();
```

Comment:
We could use `floatingCharSink` here. `LowerCaseCharSequenceIntHashMap#compute` should allocate (call `toString()`) only if the key is not present in the map.